src/interfaces.ts

```diff
@@ -182,33 +182,34 @@ export type CompilerPluginFactory = () => CompilerPlugin;
 export interface CompilerPlugin {
     name: string;
     //program events
-    beforeProgramCreate?: (builder: ProgramBuilder) => void;
-    beforePrepublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
-    afterPrepublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
-    beforePublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
-    afterPublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
-    afterProgramCreate?: (program: Program) => void;
-    beforeProgramValidate?: (program: Program) => void;
-    afterProgramValidate?: (program: Program) => void;
-    beforeProgramTranspile?: (program: Program, entries: TranspileObj[]) => void;
-    afterProgramTranspile?: (program: Program, entries: TranspileObj[]) => void;
+    beforeProgramCreate?: (builder: ProgramBuilder) => void | boolean;
+    beforePrepublish?: (builder: ProgramBuilder, files: FileObj[]) => void | boolean;
+    afterPrepublish?: (builder: ProgramBuilder, files: FileObj[]) => void | boolean;
+    beforePublish?: (builder: ProgramBuilder, files: FileObj[]) => void | boolean;
+    afterPublish?: (builder: ProgramBuilder, files: FileObj[]) => void | boolean;
+    afterProgramCreate?: (program: Program) => void | boolean;
+    beforeProgramValidate?: (program: Program) => void | boolean;
+    afterProgramValidate?: (program: Program) => void | boolean;
+    beforeProgramTranspile?: (program: Program, entries: TranspileObj[]) => void | boolean;
```

Comment:
I don't think all the events should be cancellable...

Reply:
fair enough. I can remove that from these interfaces that aren't applicable. However, we don't have a way to limit this in the actual PluginInterface right now until #330 lands (because we can add a third `isCancellable` parameter to `emit` once the second param is an event object).

Reply:
Reverted all except the `onGetHover` one.

Reply:
There could also be an `emitCancellable` in addition to `emit`

Reply:
I think for `onHover` it's not about cancelling but confirming that it's handled, as when we confirm a key event is handled.

Reply:
is it not more common to have an cancellation token?